src/Model/Behavior/UploadBehavior.php

```diff
@@ -80,6 +83,10 @@ public function beforeMarshal(Event $event, ArrayObject $data, ArrayObject $opti
     public function beforeSave(Event $event, Entity $entity, ArrayObject $options)
     {
         foreach ($this->config() as $field => $settings) {
+            if (in_array($field, $this->protectedFieldNames)) {
+                continue;
```

Comment:
We should instead raise an exception here so that developers catch this during development.

Reply:
Makes perfect sense.

Reply:
Should probably write a test too, so we know the Exception happens

Reply:
When I took a second look at it, this would make it impossible for the user to pass the priority setting to the plugin?

Reply:
Seems like the way to go. I guess this can be merged and released since it is properly documented?

Reply:
Yes I did complain about the field names being top level keys in the past. Surprised it took so long for someone to eventually face a problem due to it. 🙂